<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="Kyra Thrush" />

<meta name="date" content="2022-08-30" />

<meta name="progressive" content="true" />
<meta name="allow-skip" content="false" />
<meta name="learnr-version-prerender" content="0.10.6.9000" />

<title>methylCIPHER tutorial</title>

<!-- header-includes START -->
<!-- HEAD_CONTENT -->
<!-- header-includes END -->
<!-- HEAD_CONTENT -->

<!-- highlightjs -->
<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>


<!-- taken from https://github.com/rstudio/rmarkdown/blob/de8a9c38618903627ca509f5401d50a0876079f7/inst/rmd/h/default.html#L293-L343 -->
<!-- tabsets -->
<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>
<!-- end tabsets -->


</head>

<body>
<a class='sr-only sr-only-focusable visually-hidden-focusable' href='#learnr-tutorial-content'>Skip to Tutorial Content</a>



<div class="pageContent band">
<main class="bandContent page">

<article class="topics" id="learnr-tutorial-content">

<div id="section-introduction-background" class="section level2"
data-progressive="FALSE">
<h2>Introduction &amp; Background</h2>
<p>The goal of methylCIPHER is to allow users to easily calculate their
choice of CpG clocks using simple commands, from a single source. CpG
epigenetic clocks are currently found in many places, and some require
users to send data to external portals, which is not advisable when
working with protected or restricted data. The current package allows
you to calculate reported epigenetic clocks, or where not precisely
disclosed, our best estimates–all performed locally on your own machine
in R Studio. We would like to acknowledge the authors of the original
clocks and their valuable contributions to aging research, and the
requisite citations for their clocks can be found at the bottom of the
current page. Please do not forget to cite them in your work!</p>
<div id="section-installation" class="section level3">
<h3>Installation</h3>
<p>You can install the released version of methylCIPHER and its imported
packages from <a
href="https://Cgithub.com/MorganLevineLab/methylCIPHER">Github</a>
with:</p>
<pre class="r"><code>devtools::install_github(&quot;MorganLevineLab/prcPhenoAge&quot;)
devtools::install_github(&quot;danbelsky/DunedinPoAm38&quot;)
devtools::install_github(&quot;MorganLevineLab/methylCIPHER&quot;)</code></pre>
</div>
<div id="section-setup-your-data" class="section level3">
<h3>Setup your data</h3>
<pre class="r"><code>library(methylCIPHER)
myBetas &lt;- exampleBetas
myPheno &lt;- examplePheno</code></pre>
<p>Here we use the example data included in the methylCIPHER package,
but instead you could assign your own data to <code>myBetas</code> and
<code>myPheno</code> but using <code>load</code> for RData objects,
<code>readr::read_csv</code> or <code>read.csv</code> for those as .csv
or excel files, or whatever you like. Then just reassign the variable
names so that you can copy the commands defined throughout the rest of
this tutorial!</p>
</div>
<div id="section-categories-of-epigenetic-clocks"
class="section level3">
<h3>Categories of Epigenetic Clocks</h3>
<p>Due to the abundance of epigenetic clocks are overlapping reasons for
use, it is important to keep track of which clocks are most related to
each other. This will allow you to steer clear of multiple testing and
collinearity problems if you use all available clocks for your
analysis.</p>
<p>Another important note is that with a few exceptions, the list of
following clocks were trained and validated almost exclusively in blood.
This can lead to a number of observed effects, which include
unreasonable shifts in age (intercept)As bespoke clocks are developed
for use in additional human tissues, we will include these in their own
section below.</p>
<div id="section-chronological-age-predictors" class="section level4">
<h4>Chronological Age Predictors</h4>
<ul>
<li>Bocklandt (calcBocklandt)</li>
<li>Garagnani (calcGaragnani)</li>
<li><strong>Hannum (calcHannum)</strong></li>
<li><strong>Horvath MultiTissue (calcHorvath1)</strong></li>
<li>Lin 99 CpG clock (calcLin)</li>
<li>Vidal-Bralo (calcVidalBralo)</li>
<li>Weidner (calcWeidner)</li>
<li>Zhang 10 CpG clock (calcZhang)</li>
</ul>
</div>
<div id="section-cancer-and-mitotic-rates" class="section level4">
<h4>Cancer and Mitotic Rates</h4>
<ul>
<li>Epigenetic Timers of Cancer:
<ul>
<li>EpiTOC (calcEpiTOC)</li>
<li><strong>EpiTOC2 (calcEpiTOC2)</strong></li>
</ul></li>
<li>solo-WCGWsites, or hypoclock (calcHypoClock)</li>
<li>MiAge Calculator (calcMiAge)</li>
</ul>
</div>
<div id="section-gestational-and-pediatric-age-predictors"
class="section level4">
<h4>Gestational and Pediatric Age Predictors</h4>
<ul>
<li>Bohlin (calcBohlin)<br />
</li>
<li>Knight (calcKnight)</li>
<li>Lee Gestational Age:
<ul>
<li>calcLeeControl<br />
</li>
<li>calcLeeRobust<br />
</li>
<li>calcLeeRefinedRobust<br />
</li>
</ul></li>
<li>Mayne (calcMayne)<br />
</li>
<li>Pediatric-Buccal-Epigenetic Clock (calcPEDBE)</li>
</ul>
</div>
<div id="section-biological-age-and-mortality-predictors"
class="section level4">
<h4>Biological Age and Mortality Predictors</h4>
<ul>
<li>Dunedin Pace of Aging (DunedinPoAm38)
<ul>
<li>Updated DunedinPACE: available <a
href="https://github.com/danbelsky/DunedinPACE">elsewhere</a></li>
</ul></li>
<li><strong>PhenoAge</strong>:
<ul>
<li>Original 513 CpG PhenoAge (calcPhenoAge)
<ul>
<li>Polycomb Repressor Complex Associated Subscore (<a
href="https://github.com/MorganLevineLab/prcPhenoAge">calcprcPhenoAge</a>)</li>
<li>PRC dissociated Subscore (calcnonprcPhenoAge)</li>
</ul></li>
<li>Updated PhenoAge (calcHRSInChPhenoAge)</li>
</ul></li>
<li><strong>GrimAge:</strong> not available, use
<strong>formatHorvathOnline</strong> <a
href="http://dnamage.genetics.ucla.edu/home">Horvath Online
Calculator</a></li>
</ul>
</div>
<div id="section-trait-predictors" class="section level4">
<h4>Trait Predictors</h4>
<ul>
<li>McCartney’s Predictors:
<ul>
<li>Alcohol (calcAlcoholMcCartney)</li>
<li>BMI (calcBMIMcCartney)</li>
<li>Smoking (calcSmokingMcCartney)</li>
</ul></li>
<li>PhosphatidylEthanol: Hazardous Alcohol Use (calcPEth)</li>
</ul>
</div>
<div id="section-non-blood-clocks" class="section level4">
<h4>Non-Blood Clocks</h4>
<ul>
<li>DNAmAge<sub>Cortical</sub>
<ul>
<li><em>A brain tissue trained clock.</em></li>
</ul></li>
<li>Horvath Skin &amp; Blood (calcHorvath2)</li>
</ul>
</div>
</div>
</div>
<div id="section-selecting-epigenetic-clocks" class="section level2">
<h2>Selecting Epigenetic Clocks</h2>
<p>Ideally, the user will make a reasonably informed choice about the
epigenetic clock they should use. There are a few steps of diligence
that should be taken in order to make the best hypothesis-based
selection.</p>
<div id="section-missing-beta-values" class="section level3">
<h3>Missing beta values</h3>
<p>Of course, all of the epigenetic clocks work best when you have all
of the necessary probes’ beta values for each sample. However, sometimes
after pre-processing your data, beta values will be removed for a
variety of reasons. It is best if you select a clock, or set of clocks,
with the fewest missing values possible. To check this, we have provided
the following functions:</p>
<div class="tutorial-exercise" data-label="ProbeCheck"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0">
<pre class="text"><code>which(apply(myBetas, 2, function(x)all(is.na(x))))
# write code here to remove the all NA columns if needed...

getClockProbes(myBetas)</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<ul>
<li><p>Please note that if you have a named column of all NAs, the
getClockProbes function will not sense that those CpGs are missing.</p>
<ul>
<li>You should verify that the first line of code provides a result that
states “named integer(0)”. If not, we recommend that you remove any
identified columns from your beta matrix entirely to avoid errors, and
then recalculate the probe table.<br />
</li>
<li>If you wish to use this code chunk on your own data, simply
uncomment and provide the path to your own DNAm betas and replace
exampleBetas with “myBetas”.</li>
</ul></li>
</ul>
</div>
<div id="section-imputation" class="section level3">
<h3>Imputation</h3>
<p>For each CpG clock, you have the option to impute missing values for
CpGs that were removed across all samples. In this case, you will need
to impute using a vector of your choice (e.g. mean methylation values
across CpGs from an independent tissue-matched dataset). However, by
default, imputation will not be performed and the portion of the clock
that is reliant upon those CpGs will not be considered.</p>
<ul>
<li>In the case that you have CpGs missing from only some samples, we
encourage you to be aware of this early on. Run the following line and
check that it is 0.</li>
</ul>
<div class="tutorial-exercise" data-label="checkSporadicNA"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0">
<pre class="text"><code>sum(is.na(myBetas))</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<ul>
<li>If this does not end up being 0, you might consider running mean
imputation within your data so that NA values for single/ few samples at
least have mean values rather than being ignored.</li>
</ul>
<pre class="r"><code>exampleBetas_imputed &lt;- meanimpute(exampleBetas)</code></pre>
</div>
<div id="section-making-the-selection" class="section level3">
<h3>Making the selection</h3>
<p>Now, you should ensure that prior to calculating any clocks, you work
through these steps:</p>
<ol style="list-style-type: decimal">
<li>Develop your hypothesis you wish to measure in the data. Is it
related to measuring the chronological age of samples? The mitotic rate
of a tissue? The biological functioning of an organism or tissue with
age? Select the appropriate category.<br />
</li>
<li>Consider reviewing the papers regarding the assumptions and training
steps for the clocks within the appropriate category/ categories. The
citations to help you find the original papers can be found under the
citations section.<br />
</li>
<li>Evaluate whether there are sufficient probes in your cleaned and
normalized data to calculate your clock choice(s).</li>
</ol>
</div>
</div>
<div id="section-calculating-epigenetic-clocks" class="section level2">
<h2>Calculating Epigenetic Clocks</h2>
<div id="section-running-a-single-clock-dataframe-output"
class="section level3">
<h3>Running a single clock (Dataframe Output)</h3>
<p>The current package contains a large number of currently available
CpG clocks or CpG based predictors. While we strove to be inclusive of
such published CpG-based epigenetic clocks to our knowledge, if you find
we are missing a clock, please contact us and we will do our best to
promptly include it, if possible. You can do so by raising an issue on
this repo or emailing us directly at
&lt;kyra[dot]thrush[at]yale[dot]edu&gt;.</p>
<p>In order to calculate a CpG clock, you simply need to use the
appropriate function, typically named <code>calc[ClockNameHere]</code>.
For example:</p>
<div class="tutorial-exercise" data-label="calcPhenoAgeDF"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0">
<pre class="text"><code>calcPhenoAge(myBetas, myPheno, imputation = F)</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<p>Feel free to change the inputs and settings of this function. If you
would like to impute sporadically missing values, for example, change it
to <code>imputation = T</code>. You can also impute entirely missing
CpGs if you have a vector of reference methylation values for the
missing CpGs, and add
<code>CpGImputation = [that-vector-object]</code>.</p>
</div>
<div id="section-running-a-single-clock-vector-output"
class="section level3">
<h3>Running a single clock (Vector Output)</h3>
<p>Alternatively, if you would just like to receive a vector with the
clock values to use, rather than appending it to an existing phenotype/
demographic dataframe, simply use:</p>
<div class="tutorial-exercise" data-label="calcPhenoAgeVector"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0">
<pre class="text"><code>calcPhenoAge(myBetas, imputation = F)</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
</div>
<div id="section-running-the-core-epigenetic-clocks"
class="section level3">
<h3>Running the Core Epigenetic Clocks</h3>
<p>There are a number of epigenetic clocks available in the literature,
but the majority of studies currently focus upon comparing a select few.
Therefore, we have provided a helper function to quickly calculate these
core clocks in just one line of code. We don’t recommend relying upon
this function and encourage hypothesis-driven choices. However, this
function may be helpful for reproducing an existing paper’s results or
conducting meta-analyses across the existing literature.</p>
<div class="tutorial-exercise" data-label="calcCoreClocks"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0">
<pre class="text"><code>calcCoreClocks(myBetas, myPheno)</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div id="section-important-generating-dummy-pheno-data"
class="section level4">
<h4>Important: Generating Dummy Pheno Data</h4>
<p>Please note here that for any function provided which calculates
<em>multiple</em> epigenetic clocks (calcCoreClocks, calcUserClocks,…),
you must provide a pheno dataframe to append each clock to. If you don’t
have much information about samples, it can be something as simple as
creating a dataframe with sample names or numbers.</p>
Let’s test this. Write a line or two of code that generates and prints a
dataframe named <code>dummy</code> that has one column named
<code>IDs</code>, that contains the same sample labels as the
methylation data.
<div class="tutorial-exercise" data-label="makeDummyPheno"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":true,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise-support"
data-label="makeDummyPheno-hint-1" data-completion="1"
data-diagnostics="1" data-startover="1" data-lines="0">
<pre class="text"><code>Use data.frame to generate a dataframe and make IDs equal to the rownames of the myBetas object.</code></pre>
</div>
<div class="tutorial-exercise-support"
data-label="makeDummyPheno-solution" data-completion="1"
data-diagnostics="1" data-startover="1" data-lines="0">
<pre class="text"><code>dummy &lt;- data.frame(IDs = rownames(myBetas))</code></pre>
</div>
<p>Now that you have done this, you can use your generated dataframe to
append the clock values to. Take the core clocks function as an example,
and calculate the core clocks from <code>myBetas</code> and append the
output to your <code>dummy</code> data frame.</p>
<div class="tutorial-exercise" data-label="appendToDummy"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":true,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise-support" data-label="appendToDummy-hint-1"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0">
<pre class="text"><code>Run the calcCoreClocks function, and use the newly created `dummy` object as the datPheno input.</code></pre>
</div>
<div class="tutorial-exercise-support"
data-label="appendToDummy-solution" data-completion="1"
data-diagnostics="1" data-startover="1" data-lines="0">
<pre class="text"><code>calcCoreClocks(myBetas, dummy)</code></pre>
</div>
</div>
</div>
<div id="section-running-a-user-defined-list-of-epigenetic-clocks"
class="section level3">
<h3>Running A User-Defined List of Epigenetic Clocks</h3>
<p>The user is welcome to specify a vector of clocks that they would
like to calculate, rather than running each individual clock
calculation. In this case, you will need to choose from the following
options:</p>
<div class="tutorial-exercise" data-label="getClocks"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0">
<pre class="text"><code>clockOptions()</code></pre>
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<p>Now that you can see your options, please generate a vector of the
names of a few clocks you select (try 3!).</p>
<div class="tutorial-exercise" data-label="calcUserClocks1"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise-support"
data-label="calcUserClocks1-hint-1" data-completion="1"
data-diagnostics="1" data-startover="1" data-lines="0">
<pre class="text"><code>Make sure that you make an object that is a character vector, and includes 3 clock names!</code></pre>
</div>
<div class="tutorial-exercise-support"
data-label="calcUserClocks1-hint-2" data-completion="1"
data-diagnostics="1" data-startover="1" data-lines="0">
<pre class="text"><code>As an example, you could try `userClocks &lt;- c(&quot;calcPhenoAge&quot;,&quot;calcHorvath1&quot;,&quot;calcEpiTOC&quot;)`</code></pre>
</div>
<p>Using this <code>userClocks</code> vector, use the calcUserClocks
function to calculate all of them at once, using a pheno dataframe.</p>
<p>c(“calcPhenoAge”,“calcHorvath1”,“calcEpiTOC”)</p>
<div class="tutorial-exercise" data-label="calcUserClocks2"
data-completion="1" data-diagnostics="1" data-startover="1"
data-lines="0">
<script type="application/json" data-ui-opts="1">{"engine":"r","has_checker":false,"caption":"<span data-i18n=\"text.enginecap\" data-i18n-opts=\"{&quot;engine&quot;:&quot;R&quot;}\">R Code<\/span>"}</script>
</div>
<div class="tutorial-exercise-support"
data-label="calcUserClocks2-hint-1" data-completion="1"
data-diagnostics="1" data-startover="1" data-lines="0">
<pre class="text"><code>calcUserClocks(userClocks,...)</code></pre>
</div>
<div class="tutorial-exercise-support"
data-label="calcUserClocks2-hint-2" data-completion="1"
data-diagnostics="1" data-startover="1" data-lines="0">
<pre class="text"><code>calcUserClocks(userClocks, myBetas, ...)</code></pre>
</div>
<div class="tutorial-exercise-support"
data-label="calcUserClocks2-hint-3" data-completion="1"
data-diagnostics="1" data-startover="1" data-lines="0">
<pre class="text"><code>calcUserClocks(userClocks,myBetas, myPheno)</code></pre>
</div>
</div>
<div id="section-calculating-grimage" class="section level3">
<h3>Calculating GrimAge</h3>
<p>In order to calculate Grim Age, you will need to prepare your data to
be sent to the Horvath online Calculator. We have created a one-step
helper function to assist you with this. It is run as follows:</p>
<pre class="r"><code>formatHorvathOnline(myBetas, myPheno, writePath = &quot;~/Current-Project-Folder/&quot;)</code></pre>
<p>This will generate the formatted .csv files necessary for input to
the <a href="http://dnamage.genetics.ucla.edu/home">online Horvath
Calculator</a>.</p>
</div>
</div>
<div id="section-citations" class="section level2"
data-progressive="FALSE">
<h2>Citations!</h2>
<p>The current package is intended to increase visibility and
accesibility for the many epigenetic clocks currently in circulation.
Please do not forget to cite not only this work, but the works you use
in your research the correspond to the appropriate epigenetic
clocks.</p>
<div id="section-citemyclocks" class="section level3">
<h3>citeMyClocks()</h3>
<p>As an alternative to manually checking the list below, you can simply
use the function <strong>citeMyClocks</strong> to automatically print
out APA style citations of the appropriate publications. You can use a
character vector or input with the function name(s) you used, or the
same <em>userClocks</em> list you may have used for
<strong>calcUserClocks</strong>.</p>
<pre class="r"><code>citeMyClocks(userClocks, prettyprint = TRUE)</code></pre>
<p><em>Please note that by default the function formats the output for
nicer console output. Change to prettyprint = FALSE for a vector you can
save in your workspace.</em></p>
</div>
<div id="section-calccoreclocks-function" class="section level3">
<h3>calcCoreClocks Function</h3>
<p>If you used the <strong>calcCoreClocks</strong> function then the
list of citations you will need is as below. Otherwise, please find the
appropriate citations for the calculations you did run in their
respective sections below.</p>
<ul>
<li>Hannum, G., Guinney, J., Zhao, L., Zhang, L., Hughes, G., Sadda, S.
V., … Zhang, K. (2013). Genome-wide Methylation Profiles Reveal
Quantitative Views of Human Aging Rates. Molecular Cell, 49(2), 359–367.
<a href="https://doi.org/10.1016/j.molcel.2012.10.016"
class="uri">https://doi.org/10.1016/j.molcel.2012.10.016</a><br />
</li>
<li>Horvath, S. (2013). DNA methylation age of human tissues and cell
types. Genome Biology, 14(10), 3156. <a
href="https://doi.org/10.1186/gb-2013-14-10-r115"
class="uri">https://doi.org/10.1186/gb-2013-14-10-r115</a><br />
</li>
<li>Levine, M. E., Lu, A. T., Quach, A., Chen, B. H., Assimes, T. L.,
Bandinelli, S., … Horvath, S. (2018). An epigenetic biomarker of aging
for lifespan and healthspan. Aging, 10(4), 573–591. <a
href="https://doi.org/10.18632/aging.101414"
class="uri">https://doi.org/10.18632/aging.101414</a><br />
</li>
<li>Teschendorff, A. E. (2020). A comparison of epigenetic mitotic-like
clocks for cancer risk prediction. Genome Medicine, 12(1), 56. <a
href="https://doi.org/10.1186/s13073-020-00752-3"
class="uri">https://doi.org/10.1186/s13073-020-00752-3</a></li>
</ul>
</div>
<div id="section-chronological-age-predictors-1" class="section level3">
<h3>Chronological Age Predictors</h3>
<ul>
<li><strong>calcBocklandt:</strong> Bocklandt, S., Lin, W., Sehl, M. E.,
Sánchez, F. J., Sinsheimer, J. S., Horvath, S., &amp; Vilain, E. (2011).
Epigenetic Predictor of Age. PLOS ONE, 6(6), e14821. <a
href="https://doi.org/10.1371/journal.pone.0014821"
class="uri">https://doi.org/10.1371/journal.pone.0014821</a><br />
</li>
<li><strong>calcGaragnani:</strong> Garagnani, P., Bacalini, M. G.,
Pirazzini, C., Gori, D., Giuliani, C., Mari, D., … Franceschi, C.
(2012). Methylation of ELOVL2 gene as a new epigenetic marker of age.
Aging Cell, 11(6), 1132–1134. <a
href="https://doi.org/10.1111/acel.12005"
class="uri">https://doi.org/10.1111/acel.12005</a><br />
</li>
<li><strong>calcHannum:</strong> Hannum, G., Guinney, J., Zhao, L.,
Zhang, L., Hughes, G., Sadda, S. V., … Zhang, K. (2013). Genome-wide
Methylation Profiles Reveal Quantitative Views of Human Aging Rates.
Molecular Cell, 49(2), 359–367. <a
href="https://doi.org/10.1016/j.molcel.2012.10.016"
class="uri">https://doi.org/10.1016/j.molcel.2012.10.016</a><br />
</li>
<li><strong>calcHorvath1:</strong> Horvath, S. (2013). DNA methylation
age of human tissues and cell types. Genome Biology, 14(10), 3156. <a
href="https://doi.org/10.1186/gb-2013-14-10-r115"
class="uri">https://doi.org/10.1186/gb-2013-14-10-r115</a><br />
</li>
<li><strong>calcLin:</strong> Lin, Q., Weidner, C. I., Costa, I. G.,
Marioni, R. E., Ferreira, M. R. P., Deary, I. J., &amp; Wagner, W.
(2016). DNA methylation levels at individual age-associated CpG sites
can be indicative for life expectancy. Aging, 8(2), 394–401. <a
href="https://doi.org/10.18632/aging.100908"
class="uri">https://doi.org/10.18632/aging.100908</a><br />
</li>
<li><strong>calcVidalBralo:</strong> Vidal-Bralo, L., Lopez-Golan, Y.,
&amp; Gonzalez, A. (2016). Simplified assay for epigenetic age
estimation in whole blood of adults. Frontiers in Genetics, 7(JUL), 1–7.
<a href="https://doi.org/10.3389/fgene.2016.00126"
class="uri">https://doi.org/10.3389/fgene.2016.00126</a><br />
</li>
<li><strong>calcWeidner:</strong> Weidner, C. I., Lin, Q., Koch, C. M.,
Eisele, L., Beier, F., Ziegler, P., … Wagner, W. (2014). Aging of blood
can be tracked by DNA methylation changes at just three CpG sites.
Genome Biology, 15(2). <a
href="https://doi.org/10.1186/gb-2014-15-2-r24"
class="uri">https://doi.org/10.1186/gb-2014-15-2-r24</a><br />
</li>
<li><strong>calcZhang:</strong> Zhang, Y., Wilson, R., Heiss, J.,
Breitling, L. P., Saum, K. U., Schöttker, B., … Brenner, H. (2017). DNA
methylation signatures in peripheral blood strongly predict all-cause
mortality. Nature Communications, 8. <a
href="https://doi.org/10.1038/ncomms14617"
class="uri">https://doi.org/10.1038/ncomms14617</a><br />
</li>
<li><strong>calcZhang2019:</strong> Zhang, Q., Vallerga, C. L., Walker,
R. M., Lin, T., Henders, A. K., Montgomery, G. W., … Visscher, P. M.
(2019). Improved precision of epigenetic clock estimates across tissues
and its implication for biological ageing. Genome Medicine, 11(1), 54.
<a href="https://doi.org/10.1186/s13073-019-0667-1"
class="uri">https://doi.org/10.1186/s13073-019-0667-1</a></li>
</ul>
</div>
<div id="section-cancer-and-mitotic-rates-1" class="section level3">
<h3>Cancer and Mitotic Rates</h3>
<ul>
<li><strong>calcDNAmTL:</strong> Lu, A. T., Seeboth, A., Tsai, P. C.,
Sun, D., Quach, A., Reiner, A. P., … Horvath, S. (2019). DNA
methylation-based estimator of telomere length. Aging, 11(16),
5895–5923. <a href="https://doi.org/10.18632/aging.102173"
class="uri">https://doi.org/10.18632/aging.102173</a><br />
</li>
<li><strong>calcEpiTOC:</strong> Yang, Z., Wong, A., Kuh, D., Paul, D.
S., Rakyan, V. K., Leslie, R. D., … Teschendorff, A. E. (2016).
Correlation of an epigenetic mitotic clock with cancer risk. Genome
Biology, 17(1), 1–18. <a
href="https://doi.org/10.1186/s13059-016-1064-3"
class="uri">https://doi.org/10.1186/s13059-016-1064-3</a><br />
</li>
<li><strong>calcEpiTOC2 + calcHypoClock:</strong> Teschendorff, A. E.
(2020). A comparison of epigenetic mitotic-like clocks for cancer risk
prediction. Genome Medicine, 12(1), 56. <a
href="https://doi.org/10.1186/s13073-020-00752-3"
class="uri">https://doi.org/10.1186/s13073-020-00752-3</a><br />
</li>
<li><strong>calcMiAge:</strong> Youn, A., &amp; Wang, S. (2018). The
MiAge Calculator: a DNA methylation-based mitotic age calculator of
human tissue types. Epigenetics, 13(2), 192–206. <a
href="https://doi.org/10.1080/15592294.2017.1389361"
class="uri">https://doi.org/10.1080/15592294.2017.1389361</a></li>
</ul>
</div>
<div id="section-gestational-pediatric-age" class="section level3">
<h3>Gestational &amp; Pediatric Age</h3>
<ul>
<li><strong>calcBohlin:</strong> Bohlin, J., Håberg, S. E., Magnus, P.,
Reese, S. E., Gjessing, H. K., Magnus, M. C., … Nystad, W. (2016).
Prediction of gestational age based on genome-wide differentially
methylated regions. Genome Biology, 17(1), 1–9. <a
href="https://doi.org/10.1186/s13059-016-1063-4"
class="uri">https://doi.org/10.1186/s13059-016-1063-4</a><br />
</li>
<li><strong>calcKnight:</strong> Knight, A. K., Craig, J. M., Theda, C.,
Bækvad-Hansen, M., Bybjerg-Grauholm, J., Hansen, C. S., … Smith, A. K.
(2016). An epigenetic clock for gestational age at birth based on blood
methylation data. Genome Biology, 17(1), 1–11. <a
href="https://doi.org/10.1186/s13059-016-1068-z"
class="uri">https://doi.org/10.1186/s13059-016-1068-z</a><br />
</li>
<li>Lee, Y., Choufani, S., Weksberg, R., Wilson, S. L., Yuan, V., Burt,
A., … Horvath, S. (2019). Placental epigenetic clocks: Estimating
gestational age using placental DNA methylation levels. Aging, 11(12),
4238–4253. <a href="https://doi.org/10.18632/aging.102049"
class="uri">https://doi.org/10.18632/aging.102049</a>
<ul>
<li><strong>calcLeeControl:</strong> Control Placental Sample Age
Calculator<br />
</li>
<li><strong>calcLeeRobust:</strong> Robust predictor of Placental
Age<br />
</li>
<li><strong>calcLeeRefinedRobust:</strong> A refined version of the
Robust Predictor of Placental Age<br />
</li>
</ul></li>
<li><strong>calcMayne:</strong> Mayne, B. T., Leemaqz, S. Y., Smith, A.
K., Breen, J., Roberts, C. T., &amp; Bianco-Miotto, T. (2017).
Accelerated placental aging in early onset preeclampsia pregnancies
identified by DNA methylation. Epigenomics, 9(3), 279–289. <a
href="https://doi.org/10.2217/epi-2016-0103"
class="uri">https://doi.org/10.2217/epi-2016-0103</a><br />
</li>
<li><strong>calcPEDBE:</strong> McEwen, L. M., O’Donnell, K. J., McGill,
M. G., Edgar, R. D., Jones, M. J., MacIsaac, J. L., … Kobor, M. S.
(2020). The PedBE clock accurately estimates DNA methylation age in
pediatric buccal cells. PNAS, 117(38), 23329–23335. <a
href="https://doi.org/10.1073/pnas.1820843116"
class="uri">https://doi.org/10.1073/pnas.1820843116</a></li>
</ul>
</div>
<div id="section-biological-age-and-mortality-predictors-1"
class="section level3">
<h3>Biological Age and Mortality Predictors</h3>
<ul>
<li><strong>calcDunedinPoAm38:</strong> Belsky, D. W., Caspi, A.,
Arseneault, L., Baccarelli, A., Corcoran, D. L., Gao, X., … Moffitt, T.
E. (2020). Quantification of the pace of biological aging in humans
through a blood test, the DunedinPoAm DNA methylation algorithm. ELife,
9, e54870. <a href="https://doi.org/10.7554/eLife.54870"
class="uri">https://doi.org/10.7554/eLife.54870</a><br />
</li>
<li><strong>dunedinPACE (DunedinPACE::PoAmProjector())</strong> Belsky,
D. W., Caspi, A., Corcoran, D. L., Sugden, K., Poulton, R., Arseneault,
L., … Moffitt, T. E. (2022). DunedinPACE, a DNA methylation biomarker of
the pace of aging. ELife, 11, e73420. <a
href="https://doi.org/10.7554/eLife.73420"
class="uri">https://doi.org/10.7554/eLife.73420</a></li>
<li><strong>calcPhenoAge:</strong> Levine, M. E., Lu, A. T., Quach, A.,
Chen, B. H., Assimes, T. L., Bandinelli, S., … Horvath, S. (2018). An
epigenetic biomarker of aging for lifespan and healthspan. Aging, 10(4),
573–591. <a href="https://doi.org/10.18632/aging.101414"
class="uri">https://doi.org/10.18632/aging.101414</a><br />
</li>
<li><strong>prcPhenoAge::calcprcPhenoAge() /
calcnonprcPhenoAge()</strong> Add prcPhenoAge manuscript citation</li>
<li><strong>calcHRSInChPhenoAge:</strong> [cite PC Clocks paper
here]</li>
<li><em>GrimAge:</em> Lu, A. T., Quach, A., Wilson, J. G., Reiner, A.
P., Aviv, A., Raj, K., … Horvath, S. (2019). DNA methylation GrimAge
strongly predicts lifespan and healthspan. Aging, 11(2), 303–327. <a
href="https://doi.org/10.18632/aging.101684"
class="uri">https://doi.org/10.18632/aging.101684</a></li>
</ul>
</div>
<div id="section-trait-predictors-1" class="section level3">
<h3>Trait Predictors</h3>
<ul>
<li>McCartney, D. L., Hillary, R. F., Stevenson, A. J., Ritchie, S. J.,
Walker, R. M., Zhang, Q., … Marioni, R. E. (2018). Epigenetic prediction
of complex traits and death. Genome Biology, 19(1), 136. <a
href="https://doi.org/10.1186/s13059-018-1514-1"
class="uri">https://doi.org/10.1186/s13059-018-1514-1</a>
<ul>
<li><strong>calcAlcoholMcCartney</strong></li>
<li><strong>calcBMIMcCartney</strong></li>
<li><strong>calcSmokingMcCartney</strong></li>
</ul></li>
<li><strong>calcPEth:</strong> Liang, X., Justice, A. C., So-Armah, K.,
Krystal, J. H., Sinha, R., &amp; Xu, K. (2021). DNA methylation
signature on phosphatidylethanol, not on self-reported alcohol
consumption, predicts hazardous alcohol consumption in two distinct
populations. Molecular Psychiatry, 26(6), 2238–2253. <a
href="https://doi.org/10.1038/s41380-020-0668-x"
class="uri">https://doi.org/10.1038/s41380-020-0668-x</a></li>
</ul>
</div>
<div id="section-bespoke-clocks" class="section level3">
<h3>Bespoke Clocks</h3>
<ul>
<li><strong>calcDNAmClockCortical:</strong> Shireby, G. L., Davies, J.
P., Francis, P. T., Burrage, J., Walker, E. M., Neilson, G. W. A., …
Mill, J. (2020). Recalibrating the epigenetic clock: implications for
assessing biological age in the human cortex. Brain, 1–13. <a
href="https://doi.org/10.1093/brain/awaa334"
class="uri">https://doi.org/10.1093/brain/awaa334</a><br />
</li>
<li><strong>calcPEDBE:</strong> McEwen, L. M., O’Donnell, K. J., McGill,
M. G., Edgar, R. D., Jones, M. J., MacIsaac, J. L., … Kobor, M. S.
(2020). The PedBE clock accurately estimates DNA methylation age in
pediatric buccal cells. Proceedings of the National Academy of Sciences
of the United States of America, 117(38), 23329–23335. <a
href="https://doi.org/10.1073/pnas.1820843116"
class="uri">https://doi.org/10.1073/pnas.1820843116</a><br />
</li>
<li><strong>calcHorvath2:</strong> Horvath, S., Oshima, J., Martin, G.
M., Lu, A. T., Quach, A., Cohen, H., … Raj, K. (2018). Epigenetic clock
for skin and blood cells applied to Hutchinson Gilford Progeria Syndrome
and ex vivo studies. Aging, 10(7), 1758–1775. <a
href="https://doi.org/10.18632/aging.101508"
class="uri">https://doi.org/10.18632/aging.101508</a></li>
</ul>
<p>
<script type="application/shiny-prerendered" data-context="server-start">
knitr::opts_chunk$set(echo = TRUE)
library(learnr)
gradethis::gradethis_setup()
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::register_http_handlers(session, metadata = NULL)
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::prepare_tutorial_state(session)
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::i18n_observe_tutorial_language(input, session)
</script>


<script type="application/shiny-prerendered" data-context="server">
session$onSessionEnded(function() {
        learnr:::event_trigger(session, "session_stop")
      })
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-ProbeCheck-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-ProbeCheck-code-editor`)), session)
output$`tutorial-exercise-ProbeCheck-output` <- renderUI({
  `tutorial-exercise-ProbeCheck-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "ProbeCheck", global_setup = structure(c("knitr::opts_chunk$set(echo = TRUE)", 
"library(learnr)", "gradethis::gradethis_setup()"), chunk_opts = list(
    label = "setup", include = FALSE)), setup = "library(methylCIPHER)\nmyBetas <- exampleBetas\nmyPheno <- examplePheno", 
    chunks = list(list(label = "load-data", code = "library(methylCIPHER)\nmyBetas <- exampleBetas\nmyPheno <- examplePheno", 
        opts = list(label = "\"load-data\"", echo = "1", highlight = "TRUE"), 
        engine = "r"), list(label = "ProbeCheck", code = "\nwhich(apply(myBetas, 2, function(x)all(is.na(x))))\n# write code here to remove the all NA columns if needed...\n\ngetClockProbes(myBetas)", 
        opts = list(label = "\"ProbeCheck\"", exercise = "TRUE", 
            exercise.setup = "\"load-data\""), engine = "r")), 
    code_check = NULL, error_check = NULL, check = NULL, solution = NULL, 
    tests = NULL, options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "tutorial_cache/html/", cache.vars = NULL, 
        cache.lazy = TRUE, dependson = NULL, autodep = FALSE, 
        cache.rebuild = FALSE, fig.keep = "high", fig.show = "asis", 
        fig.align = "default", fig.path = "tutorial_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.timelimit = 60, exercise.checker = "function (label = NULL, solution_code = NULL, user_code = NULL, \n    check_code = NULL, envir_result = NULL, evaluate_result = NULL, \n    envir_prep = NULL, last_value = NULL, stage = NULL, ..., \n    solution_eval_fn = NULL) \n{\n    (utils::getFromNamespace(\"check_exercise\", \"gradethis\"))(label = label, \n        solution_code = solution_code, user_code = user_code, \n        check_code = check_code, envir_result = envir_result, \n        evaluate_result = evaluate_result, envir_prep = envir_prep, \n        last_value = last_value, stage = stage, ...)\n}", 
        exercise.error.check.code = "gradethis_error_checker()", 
        label = "ProbeCheck", exercise = TRUE, exercise.setup = "load-data", 
        code = c("", "which(apply(myBetas, 2, function(x)all(is.na(x))))", 
        "# write code here to remove the all NA columns if needed...", 
        "", "getClockProbes(myBetas)"), out.width.px = 624, out.height.px = 384, 
        params.src = "ProbeCheck, exercise = TRUE, exercise.setup= \"load-data\"", 
        fig.num = 0, exercise.df_print = "paged"), engine = "r", 
    version = "4"), class = c("r", "tutorial_exercise")))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-checkSporadicNA-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-checkSporadicNA-code-editor`)), session)
output$`tutorial-exercise-checkSporadicNA-output` <- renderUI({
  `tutorial-exercise-checkSporadicNA-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "checkSporadicNA", global_setup = structure(c("knitr::opts_chunk$set(echo = TRUE)", 
"library(learnr)", "gradethis::gradethis_setup()"), chunk_opts = list(
    label = "setup", include = FALSE)), setup = "library(methylCIPHER)\nmyBetas <- exampleBetas\nmyPheno <- examplePheno", 
    chunks = list(list(label = "load-data", code = "library(methylCIPHER)\nmyBetas <- exampleBetas\nmyPheno <- examplePheno", 
        opts = list(label = "\"load-data\"", echo = "1", highlight = "TRUE"), 
        engine = "r"), list(label = "checkSporadicNA", code = "sum(is.na(myBetas))", 
        opts = list(label = "\"checkSporadicNA\"", exercise = "TRUE", 
            exercise.setup = "\"load-data\""), engine = "r")), 
    code_check = NULL, error_check = NULL, check = NULL, solution = NULL, 
    tests = NULL, options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "tutorial_cache/html/", cache.vars = NULL, 
        cache.lazy = TRUE, dependson = NULL, autodep = FALSE, 
        cache.rebuild = FALSE, fig.keep = "high", fig.show = "asis", 
        fig.align = "default", fig.path = "tutorial_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.timelimit = 60, exercise.checker = "function (label = NULL, solution_code = NULL, user_code = NULL, \n    check_code = NULL, envir_result = NULL, evaluate_result = NULL, \n    envir_prep = NULL, last_value = NULL, stage = NULL, ..., \n    solution_eval_fn = NULL) \n{\n    (utils::getFromNamespace(\"check_exercise\", \"gradethis\"))(label = label, \n        solution_code = solution_code, user_code = user_code, \n        check_code = check_code, envir_result = envir_result, \n        evaluate_result = evaluate_result, envir_prep = envir_prep, \n        last_value = last_value, stage = stage, ...)\n}", 
        exercise.error.check.code = "gradethis_error_checker()", 
        label = "checkSporadicNA", exercise = TRUE, exercise.setup = "load-data", 
        code = "sum(is.na(myBetas))", out.width.px = 624, out.height.px = 384, 
        params.src = "checkSporadicNA, exercise = TRUE, exercise.setup= \"load-data\"", 
        fig.num = 0, exercise.df_print = "paged"), engine = "r", 
    version = "4"), class = c("r", "tutorial_exercise")))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-calcPhenoAgeDF-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-calcPhenoAgeDF-code-editor`)), session)
output$`tutorial-exercise-calcPhenoAgeDF-output` <- renderUI({
  `tutorial-exercise-calcPhenoAgeDF-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "calcPhenoAgeDF", global_setup = structure(c("knitr::opts_chunk$set(echo = TRUE)", 
"library(learnr)", "gradethis::gradethis_setup()"), chunk_opts = list(
    label = "setup", include = FALSE)), setup = "library(methylCIPHER)\nmyBetas <- exampleBetas\nmyPheno <- examplePheno", 
    chunks = list(list(label = "load-data", code = "library(methylCIPHER)\nmyBetas <- exampleBetas\nmyPheno <- examplePheno", 
        opts = list(label = "\"load-data\"", echo = "1", highlight = "TRUE"), 
        engine = "r"), list(label = "calcPhenoAgeDF", code = "calcPhenoAge(myBetas, myPheno, imputation = F)", 
        opts = list(label = "\"calcPhenoAgeDF\"", exercise = "TRUE", 
            exercise.setup = "\"load-data\""), engine = "r")), 
    code_check = NULL, error_check = NULL, check = NULL, solution = NULL, 
    tests = NULL, options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "tutorial_cache/html/", cache.vars = NULL, 
        cache.lazy = TRUE, dependson = NULL, autodep = FALSE, 
        cache.rebuild = FALSE, fig.keep = "high", fig.show = "asis", 
        fig.align = "default", fig.path = "tutorial_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.timelimit = 60, exercise.checker = "function (label = NULL, solution_code = NULL, user_code = NULL, \n    check_code = NULL, envir_result = NULL, evaluate_result = NULL, \n    envir_prep = NULL, last_value = NULL, stage = NULL, ..., \n    solution_eval_fn = NULL) \n{\n    (utils::getFromNamespace(\"check_exercise\", \"gradethis\"))(label = label, \n        solution_code = solution_code, user_code = user_code, \n        check_code = check_code, envir_result = envir_result, \n        evaluate_result = evaluate_result, envir_prep = envir_prep, \n        last_value = last_value, stage = stage, ...)\n}", 
        exercise.error.check.code = "gradethis_error_checker()", 
        label = "calcPhenoAgeDF", exercise = TRUE, exercise.setup = "load-data", 
        code = "calcPhenoAge(myBetas, myPheno, imputation = F)", 
        out.width.px = 624, out.height.px = 384, params.src = "calcPhenoAgeDF, exercise = TRUE, exercise.setup= \"load-data\"", 
        fig.num = 0, exercise.df_print = "paged"), engine = "r", 
    version = "4"), class = c("r", "tutorial_exercise")))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-calcPhenoAgeVector-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-calcPhenoAgeVector-code-editor`)), session)
output$`tutorial-exercise-calcPhenoAgeVector-output` <- renderUI({
  `tutorial-exercise-calcPhenoAgeVector-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "calcPhenoAgeVector", global_setup = structure(c("knitr::opts_chunk$set(echo = TRUE)", 
"library(learnr)", "gradethis::gradethis_setup()"), chunk_opts = list(
    label = "setup", include = FALSE)), setup = "library(methylCIPHER)\nmyBetas <- exampleBetas\nmyPheno <- examplePheno", 
    chunks = list(list(label = "load-data", code = "library(methylCIPHER)\nmyBetas <- exampleBetas\nmyPheno <- examplePheno", 
        opts = list(label = "\"load-data\"", echo = "1", highlight = "TRUE"), 
        engine = "r"), list(label = "calcPhenoAgeVector", code = "calcPhenoAge(myBetas, imputation = F)", 
        opts = list(label = "\"calcPhenoAgeVector\"", exercise = "TRUE", 
            exercise.setup = "\"load-data\""), engine = "r")), 
    code_check = NULL, error_check = NULL, check = NULL, solution = NULL, 
    tests = NULL, options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "tutorial_cache/html/", cache.vars = NULL, 
        cache.lazy = TRUE, dependson = NULL, autodep = FALSE, 
        cache.rebuild = FALSE, fig.keep = "high", fig.show = "asis", 
        fig.align = "default", fig.path = "tutorial_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.timelimit = 60, exercise.checker = "function (label = NULL, solution_code = NULL, user_code = NULL, \n    check_code = NULL, envir_result = NULL, evaluate_result = NULL, \n    envir_prep = NULL, last_value = NULL, stage = NULL, ..., \n    solution_eval_fn = NULL) \n{\n    (utils::getFromNamespace(\"check_exercise\", \"gradethis\"))(label = label, \n        solution_code = solution_code, user_code = user_code, \n        check_code = check_code, envir_result = envir_result, \n        evaluate_result = evaluate_result, envir_prep = envir_prep, \n        last_value = last_value, stage = stage, ...)\n}", 
        exercise.error.check.code = "gradethis_error_checker()", 
        label = "calcPhenoAgeVector", exercise = TRUE, exercise.setup = "load-data", 
        code = "calcPhenoAge(myBetas, imputation = F)", out.width.px = 624, 
        out.height.px = 384, params.src = "calcPhenoAgeVector, exercise = TRUE, exercise.setup= \"load-data\"", 
        fig.num = 0, exercise.df_print = "paged"), engine = "r", 
    version = "4"), class = c("r", "tutorial_exercise")))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-calcCoreClocks-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-calcCoreClocks-code-editor`)), session)
output$`tutorial-exercise-calcCoreClocks-output` <- renderUI({
  `tutorial-exercise-calcCoreClocks-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "calcCoreClocks", global_setup = structure(c("knitr::opts_chunk$set(echo = TRUE)", 
"library(learnr)", "gradethis::gradethis_setup()"), chunk_opts = list(
    label = "setup", include = FALSE)), setup = "library(methylCIPHER)\nmyBetas <- exampleBetas\nmyPheno <- examplePheno", 
    chunks = list(list(label = "load-data", code = "library(methylCIPHER)\nmyBetas <- exampleBetas\nmyPheno <- examplePheno", 
        opts = list(label = "\"load-data\"", echo = "1", highlight = "TRUE"), 
        engine = "r"), list(label = "calcCoreClocks", code = "calcCoreClocks(myBetas, myPheno)", 
        opts = list(label = "\"calcCoreClocks\"", exercise = "TRUE", 
            exercise.setup = "\"load-data\"", message = "FALSE"), 
        engine = "r")), code_check = NULL, error_check = NULL, 
    check = NULL, solution = NULL, tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "tutorial_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "tutorial_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = FALSE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.timelimit = 60, exercise.checker = "function (label = NULL, solution_code = NULL, user_code = NULL, \n    check_code = NULL, envir_result = NULL, evaluate_result = NULL, \n    envir_prep = NULL, last_value = NULL, stage = NULL, ..., \n    solution_eval_fn = NULL) \n{\n    (utils::getFromNamespace(\"check_exercise\", \"gradethis\"))(label = label, \n        solution_code = solution_code, user_code = user_code, \n        check_code = check_code, envir_result = envir_result, \n        evaluate_result = evaluate_result, envir_prep = envir_prep, \n        last_value = last_value, stage = stage, ...)\n}", 
        exercise.error.check.code = "gradethis_error_checker()", 
        label = "calcCoreClocks", exercise = TRUE, exercise.setup = "load-data", 
        code = "calcCoreClocks(myBetas, myPheno)", out.width.px = 624, 
        out.height.px = 384, params.src = "calcCoreClocks, exercise = TRUE, exercise.setup=\"load-data\", message=FALSE", 
        fig.num = 0, exercise.df_print = "paged"), engine = "r", 
    version = "4"), class = c("r", "tutorial_exercise")))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-makeDummyPheno-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-makeDummyPheno-code-editor`)), session)
output$`tutorial-exercise-makeDummyPheno-output` <- renderUI({
  `tutorial-exercise-makeDummyPheno-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "makeDummyPheno", global_setup = structure(c("knitr::opts_chunk$set(echo = TRUE)", 
"library(learnr)", "gradethis::gradethis_setup()"), chunk_opts = list(
    label = "setup", include = FALSE)), setup = "library(methylCIPHER)\nmyBetas <- exampleBetas\nmyPheno <- examplePheno", 
    chunks = list(list(label = "load-data", code = "library(methylCIPHER)\nmyBetas <- exampleBetas\nmyPheno <- examplePheno", 
        opts = list(label = "\"load-data\"", echo = "1", highlight = "TRUE"), 
        engine = "r"), list(label = "makeDummyPheno", code = "\n\n", 
        opts = list(label = "\"makeDummyPheno\"", exercise = "TRUE", 
            exercise.setup = "\"load-data\""), engine = "r")), 
    code_check = NULL, error_check = NULL, check = structure(c("grade_result(", 
    "  ", "  pass_if(~identical(.result, data.frame(IDs = rownames(myBetas))))", 
    "  ", ")"), chunk_opts = list(label = "makeDummyPheno-check")), 
    solution = structure("dummy <- data.frame(IDs = rownames(myBetas))", chunk_opts = list(
        label = "makeDummyPheno-solution")), tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "tutorial_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "tutorial_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.timelimit = 60, exercise.checker = "function (label = NULL, solution_code = NULL, user_code = NULL, \n    check_code = NULL, envir_result = NULL, evaluate_result = NULL, \n    envir_prep = NULL, last_value = NULL, stage = NULL, ..., \n    solution_eval_fn = NULL) \n{\n    (utils::getFromNamespace(\"check_exercise\", \"gradethis\"))(label = label, \n        solution_code = solution_code, user_code = user_code, \n        check_code = check_code, envir_result = envir_result, \n        evaluate_result = evaluate_result, envir_prep = envir_prep, \n        last_value = last_value, stage = stage, ...)\n}", 
        exercise.error.check.code = "gradethis_error_checker()", 
        label = "makeDummyPheno", exercise = TRUE, exercise.setup = "load-data", 
        code = c("", "", ""), out.width.px = 624, out.height.px = 384, 
        params.src = "makeDummyPheno,exercise = TRUE, exercise.setup=\"load-data\"", 
        fig.num = 0L, exercise.df_print = "paged"), engine = "r", 
    version = "4"), class = c("r", "tutorial_exercise")))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-appendToDummy-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-appendToDummy-code-editor`)), session)
output$`tutorial-exercise-appendToDummy-output` <- renderUI({
  `tutorial-exercise-appendToDummy-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "appendToDummy", global_setup = structure(c("knitr::opts_chunk$set(echo = TRUE)", 
"library(learnr)", "gradethis::gradethis_setup()"), chunk_opts = list(
    label = "setup", include = FALSE)), setup = "library(methylCIPHER)\nmyBetas <- exampleBetas\nmyPheno <- examplePheno\ndummy <- data.frame(IDs = rownames(myBetas))", 
    chunks = list(list(label = "load-data", code = "library(methylCIPHER)\nmyBetas <- exampleBetas\nmyPheno <- examplePheno", 
        opts = list(label = "\"load-data\"", echo = "1", highlight = "TRUE"), 
        engine = "r"), list(label = "makeDummyAnswer", code = "dummy <- data.frame(IDs = rownames(myBetas))", 
        opts = list(label = "\"makeDummyAnswer\"", exercise.setup = "\"load-data\""), 
        engine = "r"), list(label = "appendToDummy", code = "", 
        opts = list(label = "\"appendToDummy\"", exercise = "TRUE", 
            exercise.setup = "\"makeDummyAnswer\"", message = "FALSE"), 
        engine = "r")), code_check = NULL, error_check = NULL, 
    check = structure(c("grade_result(", "  ", "  pass_if(~identical(.result, calcCoreClocks(myBetas, dummy)))", 
    "  ", ")"), chunk_opts = list(label = "appendToDummy-check")), 
    solution = structure("calcCoreClocks(myBetas, dummy)", chunk_opts = list(
        label = "appendToDummy-solution")), tests = NULL, options = list(
        eval = FALSE, echo = TRUE, results = "markup", tidy = FALSE, 
        tidy.opts = NULL, collapse = FALSE, prompt = FALSE, comment = NA, 
        highlight = FALSE, size = "normalsize", background = "#F7F7F7", 
        strip.white = TRUE, cache = 0, cache.path = "tutorial_cache/html/", 
        cache.vars = NULL, cache.lazy = TRUE, dependson = NULL, 
        autodep = FALSE, cache.rebuild = FALSE, fig.keep = "high", 
        fig.show = "asis", fig.align = "default", fig.path = "tutorial_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = FALSE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.timelimit = 60, exercise.checker = "function (label = NULL, solution_code = NULL, user_code = NULL, \n    check_code = NULL, envir_result = NULL, evaluate_result = NULL, \n    envir_prep = NULL, last_value = NULL, stage = NULL, ..., \n    solution_eval_fn = NULL) \n{\n    (utils::getFromNamespace(\"check_exercise\", \"gradethis\"))(label = label, \n        solution_code = solution_code, user_code = user_code, \n        check_code = check_code, envir_result = envir_result, \n        evaluate_result = evaluate_result, envir_prep = envir_prep, \n        last_value = last_value, stage = stage, ...)\n}", 
        exercise.error.check.code = "gradethis_error_checker()", 
        label = "appendToDummy", exercise = TRUE, exercise.setup = "makeDummyAnswer", 
        code = "", out.width.px = 624, out.height.px = 384, params.src = "appendToDummy, exercise = TRUE, exercise.setup = \"makeDummyAnswer\", message=FALSE", 
        fig.num = 0L, exercise.df_print = "paged"), engine = "r", 
    version = "4"), class = c("r", "tutorial_exercise")))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-getClocks-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-getClocks-code-editor`)), session)
output$`tutorial-exercise-getClocks-output` <- renderUI({
  `tutorial-exercise-getClocks-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "getClocks", global_setup = structure(c("knitr::opts_chunk$set(echo = TRUE)", 
"library(learnr)", "gradethis::gradethis_setup()"), chunk_opts = list(
    label = "setup", include = FALSE)), setup = NULL, chunks = list(
    list(label = "getClocks", code = "clockOptions()", opts = list(
        label = "\"getClocks\"", exercise = "TRUE"), engine = "r")), 
    code_check = NULL, error_check = NULL, check = NULL, solution = NULL, 
    tests = NULL, options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "tutorial_cache/html/", cache.vars = NULL, 
        cache.lazy = TRUE, dependson = NULL, autodep = FALSE, 
        cache.rebuild = FALSE, fig.keep = "high", fig.show = "asis", 
        fig.align = "default", fig.path = "tutorial_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.timelimit = 60, exercise.checker = "function (label = NULL, solution_code = NULL, user_code = NULL, \n    check_code = NULL, envir_result = NULL, evaluate_result = NULL, \n    envir_prep = NULL, last_value = NULL, stage = NULL, ..., \n    solution_eval_fn = NULL) \n{\n    (utils::getFromNamespace(\"check_exercise\", \"gradethis\"))(label = label, \n        solution_code = solution_code, user_code = user_code, \n        check_code = check_code, envir_result = envir_result, \n        evaluate_result = evaluate_result, envir_prep = envir_prep, \n        last_value = last_value, stage = stage, ...)\n}", 
        exercise.error.check.code = "gradethis_error_checker()", 
        label = "getClocks", exercise = TRUE, code = "clockOptions()", 
        out.width.px = 624, out.height.px = 384, params.src = "getClocks, exercise = TRUE", 
        fig.num = 0, exercise.df_print = "paged"), engine = "r", 
    version = "4"), class = c("r", "tutorial_exercise")))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-calcUserClocks1-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-calcUserClocks1-code-editor`)), session)
output$`tutorial-exercise-calcUserClocks1-output` <- renderUI({
  `tutorial-exercise-calcUserClocks1-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "calcUserClocks1", global_setup = structure(c("knitr::opts_chunk$set(echo = TRUE)", 
"library(learnr)", "gradethis::gradethis_setup()"), chunk_opts = list(
    label = "setup", include = FALSE)), setup = NULL, chunks = list(
    list(label = "calcUserClocks1", code = "", opts = list(label = "\"calcUserClocks1\"", 
        exercise = "TRUE"), engine = "r")), code_check = NULL, 
    error_check = NULL, check = NULL, solution = NULL, tests = NULL, 
    options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "tutorial_cache/html/", cache.vars = NULL, 
        cache.lazy = TRUE, dependson = NULL, autodep = FALSE, 
        cache.rebuild = FALSE, fig.keep = "high", fig.show = "asis", 
        fig.align = "default", fig.path = "tutorial_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.timelimit = 60, exercise.checker = "function (label = NULL, solution_code = NULL, user_code = NULL, \n    check_code = NULL, envir_result = NULL, evaluate_result = NULL, \n    envir_prep = NULL, last_value = NULL, stage = NULL, ..., \n    solution_eval_fn = NULL) \n{\n    (utils::getFromNamespace(\"check_exercise\", \"gradethis\"))(label = label, \n        solution_code = solution_code, user_code = user_code, \n        check_code = check_code, envir_result = envir_result, \n        evaluate_result = evaluate_result, envir_prep = envir_prep, \n        last_value = last_value, stage = stage, ...)\n}", 
        exercise.error.check.code = "gradethis_error_checker()", 
        label = "calcUserClocks1", exercise = TRUE, code = "", 
        out.width.px = 624, out.height.px = 384, params.src = "calcUserClocks1, exercise = TRUE", 
        fig.num = 0L, exercise.df_print = "paged"), engine = "r", 
    version = "4"), class = c("r", "tutorial_exercise")))
</script>
 
<script type="application/shiny-prerendered" data-context="server">
`tutorial-exercise-calcUserClocks2-result` <- learnr:::setup_exercise_handler(reactive(req(input$`tutorial-exercise-calcUserClocks2-code-editor`)), session)
output$`tutorial-exercise-calcUserClocks2-output` <- renderUI({
  `tutorial-exercise-calcUserClocks2-result`()
})
</script>


<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_cache(structure(list(label = "calcUserClocks2", global_setup = structure(c("knitr::opts_chunk$set(echo = TRUE)", 
"library(learnr)", "gradethis::gradethis_setup()"), chunk_opts = list(
    label = "setup", include = FALSE)), setup = "library(methylCIPHER)\nmyBetas <- exampleBetas\nmyPheno <- examplePheno\n\nuserClocks <- c(\"calcPhenoAge\",\"calcHorvath1\",\"calcEpiTOC\")\n", 
    chunks = list(list(label = "load-data", code = "library(methylCIPHER)\nmyBetas <- exampleBetas\nmyPheno <- examplePheno", 
        opts = list(label = "\"load-data\"", echo = "1", highlight = "TRUE"), 
        engine = "r"), list(label = "calcUserClocks2Setup", code = "\nuserClocks <- c(\"calcPhenoAge\",\"calcHorvath1\",\"calcEpiTOC\")\n", 
        opts = list(label = "\"calcUserClocks2Setup\"", exercise.setup = "\"load-data\""), 
        engine = "r"), list(label = "calcUserClocks2", code = "", 
        opts = list(label = "\"calcUserClocks2\"", exercise = "TRUE", 
            exercise.setup = "\"calcUserClocks2Setup\""), engine = "r")), 
    code_check = NULL, error_check = NULL, check = NULL, solution = NULL, 
    tests = NULL, options = list(eval = FALSE, echo = TRUE, results = "markup", 
        tidy = FALSE, tidy.opts = NULL, collapse = FALSE, prompt = FALSE, 
        comment = NA, highlight = FALSE, size = "normalsize", 
        background = "#F7F7F7", strip.white = TRUE, cache = 0, 
        cache.path = "tutorial_cache/html/", cache.vars = NULL, 
        cache.lazy = TRUE, dependson = NULL, autodep = FALSE, 
        cache.rebuild = FALSE, fig.keep = "high", fig.show = "asis", 
        fig.align = "default", fig.path = "tutorial_files/figure-html/", 
        dev = "png", dev.args = NULL, dpi = 192, fig.ext = "png", 
        fig.width = 6.5, fig.height = 4, fig.env = "figure", 
        fig.cap = NULL, fig.scap = NULL, fig.lp = "fig:", fig.subcap = NULL, 
        fig.pos = "", out.width = 624, out.height = NULL, out.extra = NULL, 
        fig.retina = 2, external = TRUE, sanitize = FALSE, interval = 1, 
        aniopts = "controls,loop", warning = TRUE, error = FALSE, 
        message = TRUE, render = NULL, ref.label = NULL, child = NULL, 
        engine = "r", split = FALSE, include = TRUE, purl = TRUE, 
        max.print = 1000, exercise.timelimit = 60, exercise.checker = "function (label = NULL, solution_code = NULL, user_code = NULL, \n    check_code = NULL, envir_result = NULL, evaluate_result = NULL, \n    envir_prep = NULL, last_value = NULL, stage = NULL, ..., \n    solution_eval_fn = NULL) \n{\n    (utils::getFromNamespace(\"check_exercise\", \"gradethis\"))(label = label, \n        solution_code = solution_code, user_code = user_code, \n        check_code = check_code, envir_result = envir_result, \n        evaluate_result = evaluate_result, envir_prep = envir_prep, \n        last_value = last_value, stage = stage, ...)\n}", 
        exercise.error.check.code = "gradethis_error_checker()", 
        label = "calcUserClocks2", exercise = TRUE, exercise.setup = "calcUserClocks2Setup", 
        code = "", out.width.px = 624, out.height.px = 384, params.src = "calcUserClocks2, exercise = TRUE, exercise.setup=\"calcUserClocks2Setup\"", 
        fig.num = 0L, exercise.df_print = "paged"), engine = "r", 
    version = "4"), class = c("r", "tutorial_exercise")))
</script>
</p>
<!--html_preserve-->
<script type="application/shiny-prerendered" data-context="dependencies">
{"type":"list","attributes":{},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["header-attrs"]},{"type":"character","attributes":{},"value":["2.16"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/pandoc"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["header-attrs.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["2.16"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["jquery"]},{"type":"character","attributes":{},"value":["3.6.0"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/3.6.0"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery-3.6.0.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquerylib"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.1.4"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["bootstrap"]},{"type":"character","attributes":{},"value":["3.3.5"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/bootstrap"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["viewport"]}},"value":[{"type":"character","attributes":{},"value":["width=device-width, initial-scale=1"]}]},{"type":"character","attributes":{},"value":["js/bootstrap.min.js","shim/html5shiv.min.js","shim/respond.min.js"]},{"type":"character","attributes":{},"value":["css/cerulean.min.css"]},{"type":"character","attributes":{},"value":["<style>h1 {font-size: 34px;}\n       h1.title {font-size: 38px;}\n       h2 {font-size: 30px;}\n       h3 {font-size: 24px;}\n       h4 {font-size: 18px;}\n       h5 {font-size: 16px;}\n       h6 {font-size: 12px;}\n       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}\n       pre:not([class]) { background-color: white }<\/style>"]},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["2.16"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["pagedtable"]},{"type":"character","attributes":{},"value":["1.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/pagedtable-1.1"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["js/pagedtable.js"]},{"type":"character","attributes":{},"value":["css/pagedtable.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["2.16"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["highlightjs"]},{"type":"character","attributes":{},"value":["9.12.0"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/highlightjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["highlight.js"]},{"type":"character","attributes":{},"value":["textmate.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["2.16"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["tutorial"]},{"type":"character","attributes":{},"value":["0.10.6.9000"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/tutorial"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["tutorial.js"]},{"type":"character","attributes":{},"value":["tutorial.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.10.6.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["i18n"]},{"type":"character","attributes":{},"value":["21.6.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/i18n"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["i18next.min.js","tutorial-i18n-init.js"]},{"type":"NULL"},{"type":"character","attributes":{},"value":["<script id=\"i18n-cstm-trns\" type=\"application/json\">{\"language\":\"en\",\"resources\":{\"en\":{\"translation\":{\"button\":{\"runcode\":\"Run Code\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"Hint\",\"hint_plural\":\"Hints\",\"hinttitle\":\"$t(button.hint)\",\"hintnext\":\"Next Hint\",\"hintprev\":\"Previous Hint\",\"solution\":\"Solution\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"Copy to Clipboard\",\"startover\":\"Start Over\",\"startovertitle\":\"$t(button.startover)\",\"continue\":\"Continue\",\"submitanswer\":\"Submit Answer\",\"submitanswertitle\":\"$t(button.submitanswer)\",\"previoustopic\":\"Previous Topic\",\"nexttopic\":\"Next Topic\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"Try Again\"},\"text\":{\"startover\":\"Start Over\",\"areyousure\":\"Are you sure you want to start over? (all exercise progress will be reset)\",\"youmustcomplete\":\"You must complete the\",\"exercise\":\"exercise\",\"exercise_plural\":\"exercises\",\"inthissection\":\"in this section before continuing.\",\"code\":\"Code\",\"enginecap\":\"{{engine}} $t(text.code)\",\"quiz\":\"Quiz\",\"blank\":\"blank\",\"blank_plural\":\"blanks\",\"exercisecontainsblank\":\"This exercise contains {{count}} $t(text.blank).\",\"pleasereplaceblank\":\"Please replace {{blank}} with valid code.\",\"unparsable\":\"It looks like this might not be valid R code. R cannot determine how to turn your text into a complete command. You may have forgotten to fill in a blank, to remove an underscore, to include a comma between arguments, or to close an opening <code>&quot;<\\/code>, <code>'<\\/code>, <code>(<\\/code> or <code>{<\\/code> with a matching <code>&quot;<\\/code>, <code>'<\\/code>, <code>)<\\/code> or <code>}<\\/code>.\\n\",\"unparsablequotes\":\"<p>It looks like your R code contains specially formatted quotation marks or &quot;curly&quot; quotes (<code>{{character}}<\\/code>) around character strings, making your code invalid. R requires character values to be contained in straight quotation marks (<code>&quot;<\\/code> or <code>'<\\/code>).<\\/p> {{code}} <p>Don't worry, this is a common source of errors when you copy code from another app that applies its own formatting to text. You can try replacing the code on that line with the following. There may be other places that need to be fixed, too.<\\/p> {{suggestion}}\\n\",\"unparsableunicode\":\"<p>It looks like your R code contains an unexpected special character (<code>{{character}}<\\/code>) that makes your code invalid.<\\/p> {{code}} <p>Sometimes your code may contain a special character that looks like a regular character, especially if you copy and paste the code from another app. Try deleting the special character from your code and retyping it manually.<\\/p>\\n\",\"unparsableunicodesuggestion\":\"<p>It looks like your R code contains an unexpected special character (<code>{{character}}<\\/code>) that makes your code invalid.<\\/p> {{code}} <p>Sometimes your code may contain a special character that looks like a regular character, especially if you copy and paste the code from another app. You can try replacing the code on that line with the following. There may be other places that need to be fixed, too.<\\/p> {{suggestion}}\\n\",\"and\":\"and\",\"or\":\"or\",\"listcomma\":\", \",\"oxfordcomma\":\",\"}}},\"fr\":{\"translation\":{\"button\":{\"runcode\":\"Lancer le Code\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"Indication\",\"hint_plural\":\"Indications\",\"hinttitle\":\"$t(button.hint)\",\"hintnext\":\"Indication Suivante\",\"hintprev\":\"Indication Précédente\",\"solution\":\"Solution\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"Copier dans le Presse-papier\",\"startover\":\"Recommencer\",\"startovertitle\":\"$t(button.startover)\",\"continue\":\"Continuer\",\"submitanswer\":\"Soumettre\",\"submitanswertitle\":\"$t(button.submitanswer)\",\"previoustopic\":\"Chapitre Précédent\",\"nexttopic\":\"Chapitre Suivant\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"Réessayer\"},\"text\":{\"startover\":\"Recommencer\",\"areyousure\":\"Êtes-vous certains de vouloir recommencer? (La progression sera remise à zéro)\",\"youmustcomplete\":\"Vous devez d'abord compléter\",\"exercise\":\"l'exercice\",\"exercise_plural\":\"des exercices\",\"inthissection\":\"de cette section avec de continuer.\",\"code\":\"Code\",\"enginecap\":\"$t(text.code) {{engine}}\",\"quiz\":\"Quiz\",\"and\":\"et\",\"or\":\"ou\",\"oxfordcomma\":\"\"}}},\"es\":{\"translation\":{\"button\":{\"runcode\":\"Ejecutar código\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"Pista\",\"hint_plural\":\"Pistas\",\"hinttitle\":\"$t(button.hint)\",\"hintnext\":\"Siguiente pista\",\"hintprev\":\"Pista anterior\",\"solution\":\"Solución\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"Copiar al portapapeles\",\"startover\":\"Reiniciar\",\"startovertitle\":\"$t(button.startover)\",\"continue\":\"Continuar\",\"submitanswer\":\"Enviar respuesta\",\"submitanswertitle\":\"$t(button.submitanswer)\",\"previoustopic\":\"Tema anterior\",\"nexttopic\":\"Tema siguiente\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"Volver a intentar\"},\"text\":{\"startover\":\"Reiniciar\",\"areyousure\":\"¿De verdad quieres empezar de nuevo? (todo el progreso del ejercicio se perderá)\",\"youmustcomplete\":\"Debes completar\",\"exercise\":\"el ejercicio\",\"exercise_plural\":\"los ejercicios\",\"inthissection\":\"en esta sección antes de continuar.\",\"code\":\"Código\",\"enginecap\":\"$t(text.code) {{engine}}\",\"quiz\":\"Cuestionario\",\"and\":\"y\",\"or\":\"o\",\"oxfordcomma\":\"\"}}},\"pt\":{\"translation\":{\"button\":{\"runcode\":\"Executar código\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"Dica\",\"hint_plural\":\"Dicas\",\"hinttitle\":\"$t(button.hint)\",\"hintnext\":\"Próxima dica\",\"hintprev\":\"Dica anterior\",\"solution\":\"Solução\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"Copiar para a área de transferência\",\"startover\":\"Reiniciar\",\"startovertitle\":\"$t(button.startover)\",\"continue\":\"Continuar\",\"submitanswer\":\"Enviar resposta\",\"submitanswertitle\":\"$t(button.submitanswer)\",\"previoustopic\":\"Tópico anterior\",\"nexttopic\":\"Próximo tópico\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"Tentar novamente\"},\"text\":{\"startover\":\"Reiniciar\",\"areyousure\":\"Tem certeza que deseja começar novamente? (todo o progresso feito será perdido)\",\"youmustcomplete\":\"Você deve completar\",\"exercise\":\"o exercício\",\"exercise_plural\":\"os exercícios\",\"inthissection\":\"nesta seção antes de continuar.\",\"code\":\"Código\",\"enginecap\":\"$t(text.code) {{engine}}\",\"quiz\":\"Quiz\",\"and\":\"e\",\"or\":\"ou\",\"oxfordcomma\":\"\"}}},\"tr\":{\"translation\":{\"button\":{\"runcode\":\"Çalıştırma Kodu\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"Ipucu\",\"hint_plural\":\"İpuçları\",\"hinttitle\":\"$t(button.hint)\",\"hintnext\":\"Sonraki İpucu\",\"hintprev\":\"Önceki İpucu\",\"solution\":\"Çözüm\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"Pano'ya Kopyala\",\"startover\":\"Baştan Başlamak\",\"startovertitle\":\"$t(button.startover)\",\"continue\":\"Devam et\",\"submitanswer\":\"Cevabı onayla\",\"submitanswertitle\":\"$t(button.submitanswer)\",\"previoustopic\":\"Önceki Konu\",\"nexttopic\":\"Sonraki Konu\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"Tekrar Deneyin\"},\"text\":{\"startover\":\"Baştan Başlamak\",\"areyousure\":\"Baştan başlamak istediğinizden emin misiniz? (tüm egzersiz ilerlemesi kaybolacak)\",\"youmustcomplete\":\"Tamamlamalısın\",\"exercise\":\"egzersiz\",\"exercise_plural\":\"egzersizler\",\"inthissection\":\"devam etmeden önce bu bölümde\",\"code\":\"Kod\",\"enginecap\":\"$t(text.code) {{engine}}\",\"quiz\":\"Sınav\",\"oxfordcomma\":\"\"}}},\"emo\":{\"translation\":{\"button\":{\"runcode\":\"🏃\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"💡\",\"hint_plural\":\"$t(button.hint)\",\"hinttitle\":\"$t(button.hint)\",\"solution\":\"🎯\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"📋\",\"startover\":\"⏮\",\"startovertitle\":\"Start Over\",\"continue\":\"✅\",\"submitanswer\":\"🆗\",\"submitanswertitle\":\"Submit Answer\",\"previoustopic\":\"⬅\",\"nexttopic\":\"➡\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"🔁\"},\"text\":{\"startover\":\"⏮\",\"areyousure\":\"🤔\",\"youmustcomplete\":\"⚠️ 👉 🧑‍💻\",\"exercise\":\"\",\"exercise_plural\":\"\",\"inthissection\":\"\",\"code\":\"💻\",\"enginecap\":\"$t(text.code) {{engine}}\",\"oxfordcomma\":\"\"}}},\"eu\":{\"translation\":{\"button\":{\"runcode\":\"Kodea egikaritu\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"Laguntza\",\"hint_plural\":\"Laguntza\",\"hinttitle\":\"$t(button.hint)\",\"hintnext\":\"Aurreko laguntza\",\"hintprev\":\"Hurrengo laguntza\",\"solution\":\"Ebazpena\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"Arbelean kopiatu\",\"startover\":\"Berrabiarazi\",\"startovertitle\":\"$t(button.startover)\",\"continue\":\"Jarraitu\",\"submitanswer\":\"Erantzuna bidali\",\"submitanswertitle\":\"$t(button.submitanswer)\",\"previoustopic\":\"Aurreko atala\",\"nexttopic\":\"Hurrengo atala\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"Berriro saiatu\"},\"text\":{\"startover\":\"Berrabiarazi\",\"areyousure\":\"Berriro hasi nahi duzu? (egindako lana galdu egingo da)\",\"youmustcomplete\":\"Aurrera egin baino lehen atal honetako\",\"exercise\":\"ariketa egin behar duzu.\",\"exercise_plural\":\"ariketak egin behar dituzu.\",\"inthissection\":\"\",\"code\":\"Kodea\",\"enginecap\":\"$t(text.code) {{engine}}\",\"quiz\":\"Galdetegia\",\"oxfordcomma\":\"\"}}},\"de\":{\"translation\":{\"button\":{\"runcode\":\"Code ausführen\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"Tipp\",\"hint_plural\":\"Tipps\",\"hinttitle\":\"$t(button.hint)\",\"hintnext\":\"Nächster Tipp\",\"hintprev\":\"Vorheriger Tipp\",\"solution\":\"Lösung\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"In die Zwischenablage kopieren\",\"startover\":\"Neustart\",\"startovertitle\":\"$t(button.startover)\",\"continue\":\"Weiter\",\"submitanswer\":\"Antwort einreichen\",\"submitanswertitle\":\"$t(button.submitanswer)\",\"previoustopic\":\"Vorheriges Kapitel\",\"nexttopic\":\"Nächstes Kapitel\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"Nochmal versuchen\"},\"text\":{\"startover\":\"Neustart\",\"areyousure\":\"Bist du sicher, dass du neustarten willst? (der gesamte Lernfortschritt wird gelöscht)\",\"youmustcomplete\":\"Vervollstädinge\",\"exercise\":\"die Übung\",\"exercise_plural\":\"die Übungen\",\"inthissection\":\"in diesem Kapitel, bevor du fortfährst.\",\"code\":\"Code\",\"enginecap\":\"$t(text.code) {{engine}}\",\"quiz\":\"Quiz\",\"blank\":\"Lücke\",\"blank_plural\":\"Lücken\",\"pleasereplaceblank\":\"Bitte ersetze {{blank}} mit gültigem Code.\",\"unparsable\":\"Dies scheint kein gültiger R Code zu sein. R kann deinen Text nicht in einen gültigen Befehl übersetzen. Du hast vielleicht vergessen, die Lücke zu füllen, einen Unterstrich zu entfernen, ein Komma zwischen Argumente zu setzen oder ein eröffnendes <code>&quot;<\\/code>, <code>'<\\/code>, <code>(<\\/code> oder <code>{<\\/code> mit einem zugehörigen <code>&quot;<\\/code>, <code>'<\\/code>, <code>)<\\/code> oder <code>}<\\/code> zu schließen.\\n\",\"and\":\"und\",\"or\":\"oder\",\"listcomma\":\", \",\"oxfordcomma\":\",\"}}},\"ko\":{\"translation\":{\"button\":{\"runcode\":\"코드 실행\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"힌트\",\"hint_plural\":\"힌트들\",\"hinttitle\":\"$t(button.hint)\",\"hintnext\":\"다음 힌트\",\"hintprev\":\"이전 힌트\",\"solution\":\"솔루션\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"클립보드에 복사\",\"startover\":\"재학습\",\"startovertitle\":\"$t(button.startover)\",\"continue\":\"다음 학습으로\",\"submitanswer\":\"정답 제출\",\"submitanswertitle\":\"$t(button.submitanswer)\",\"previoustopic\":\"이전 토픽\",\"nexttopic\":\"다음 토픽\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"재시도\"},\"text\":{\"startover\":\"재학습\",\"areyousure\":\"다시 시작 하시겠습니까? (모든 예제의 진행 정보가 재설정됩니다)\",\"youmustcomplete\":\"당신은 완료해야 합니다\",\"exercise\":\"연습문제\",\"exercise_plural\":\"연습문제들\",\"inthissection\":\"이 섹션을 실행하기 전에\",\"code\":\"코드\",\"enginecap\":\"$t(text.code) {{engine}}\",\"quiz\":\"퀴즈\",\"blank\":\"공백\",\"blank_plural\":\"공백들\",\"exercisecontainsblank\":\"이 연습문제에는 {{count}}개의 $t(text.blank)이 포함되어 있습니다.\",\"pleasereplaceblank\":\"{{blank}}를 유효한 코드로 바꾸십시오.\",\"unparsable\":\"이것은 유효한 R 코드가 아닐 수 있습니다. R은 텍스트를 완전한 명령으로 변환하는 방법을 결정할 수 없습니다. 당신은 공백이나 밑줄을 대체하여 채우기, 인수를 컴마로 구분하기, 또는 <code>&quot;<\\/code>, <code>'<\\/code>, <code>(<\\/code> , <code>{<\\/code>로 시작하는 구문을 닫는 <code>&quot;<\\/code>, <code>'<\\/code>, <code>)<\\/code>, <code>}<\\/code>을 잊었을 수도 있습니다.\\n\",\"and\":\"그리고\",\"or\":\"혹은\",\"listcomma\":\", \",\"oxfordcomma\":\"\"}}},\"zh\":{\"translation\":{\"button\":{\"runcode\":\"运行代码\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"提示\",\"hint_plural\":\"提示\",\"hinttitle\":\"$t(button.hint)\",\"hintnext\":\"下一个提示\",\"hintprev\":\"上一个提示\",\"solution\":\"答案\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"复制到剪切板\",\"startover\":\"重新开始\",\"startovertitle\":\"$t(button.startover)\",\"continue\":\"继续\",\"submitanswer\":\"提交答案\",\"submitanswertitle\":\"$t(button.submitanswer)\",\"previoustopic\":\"上一专题\",\"nexttopic\":\"下一专题\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"再试一次\"},\"text\":{\"startover\":\"重置\",\"areyousure\":\"你确定要重新开始吗? (所有当前进度将被重置)\",\"youmustcomplete\":\"你必须完成\",\"exercise\":\"练习\",\"exercise_plural\":\"练习\",\"inthissection\":\"在进行本节之前\",\"code\":\"代码\",\"enginecap\":\"$t(text.code) {{engine}}\",\"quiz\":\"测试\",\"blank\":\"空\",\"blank_plural\":\"空\",\"exercisecontainsblank\":\"本练习包含{{count}}个$t(text.blank)\",\"pleasereplaceblank\":\"请在{{blank}}内填写恰当的代码\",\"unparsable\":\"这似乎不是有效的R代码。 R不知道如何将您的文本转换为完整的命令。 您是否忘了填空，忘了删除下划线，忘了在参数之间包含逗号，或者是忘了用<code>&quot;<\\/code>, <code>'<\\/code>, <code>)<\\/code>,<code>}<\\/code>来封闭<code>&quot;<\\/code>, <code>'<\\/code>, <code>(<\\/code>。 or <code>{<\\/code>。\\n\",\"unparsablequotes\":\"<p>您的R代码中似乎含有特殊格式的引号，或者弯引号(<code>{{character}}<\\/code>) 在字符串前后，在R中字符串应该被直引号(<code>&quot;<\\/code> 或者 <code>'<\\/code>)包裹。<\\/p> {{code}} <p>别担心，该错误经常在复制粘贴包含格式的代码时遇到， 您可以尝试将该行中的代码替换为以下代码，也许还有其他地方需要修改。<\\/p> {{suggestion}}\\n\",\"unparsableunicode\":\"<p>您的代码中似乎包含有异常字符(<code>{{character}}<\\/code>),导致代码无效。<\\/p> {{code}} <p>有时候你的代码可能含有看似正常字符的特殊字符，特别是当你复制粘贴其他来源代码的时候。 请试着删除这些特殊字符,重新输入<\\/p>\\n\",\"unparsableunicodesuggestion\":\"<p>您的代码中似乎包含有异常字符(<code>{{character}}<\\/code>),导致代码无效。<\\/p> {{code}} <p>有时候你的代码可能含有看似正常字符的特殊字符，特别是当你复制粘贴其他来源代码的时候。 请试着删除这些特殊字符,重新输入<\\/p>\\n\",\"and\":\"且\",\"or\":\"或\",\"listcomma\":\",\",\"oxfordcomma\":\",\"}}},\"pl\":{\"translation\":{\"button\":{\"runcode\":\"Uruchom kod\",\"runcodetitle\":\"$t(button.runcode) ({{kbd}})\",\"hint\":\"Podpowiedź\",\"hint_plural\":\"Podpowiedzi\",\"hinttitle\":\"$t(button.hint)\",\"hintnext\":\"Następna podpowiedź\",\"hintprev\":\"Poprzednia podpowiedź\",\"solution\":\"Rozwiązanie\",\"solutiontitle\":\"$t(button.solution)\",\"copyclipboard\":\"Kopiuj do schowka\",\"startover\":\"Zacznij od początku\",\"startovertitle\":\"$t(button.startover)\",\"continue\":\"Kontynuuj\",\"submitanswer\":\"Wyślij\",\"submitanswertitle\":\"$t(button.submitanswer)\",\"previoustopic\":\"Poprzednia sekcja\",\"nexttopic\":\"Następna sekcja\",\"questionsubmit\":\"$t(button.submitanswer)\",\"questiontryagain\":\"Spróbuj ponownie\"},\"text\":{\"startover\":\"Zacznij od początku\",\"areyousure\":\"Czy na pewno chcesz zacząć od początku? (cały postęp w zadaniu zostanie utracony)\",\"youmustcomplete\":\"Musisz ukończyć\",\"exercise\":\"ćwiczenie\",\"exercise_plural\":\"ćwiczenia\",\"inthissection\":\"w tej sekcji przed kontynuowaniem\",\"code\":\"Kod\",\"enginecap\":\"$t(text.code) {{engine}}\",\"quiz\":\"Quiz\",\"blank\":\"luka\",\"blank_plural\":\"luk(i)\",\"exercisecontainsblank\":\"To ćwiczenie zawiera {{count}} $t(text.blank).\",\"pleasereplaceblank\":\"Proszę uzupełnić {{blank}} prawidłowym kodem.\",\"unparsable\":\"Wygląda na to, że może to nie być prawidłowy kod R. R nie jest w stanie przetworzyć Twojego tekstu na polecenie. Mogłeś(-aś) zapomnieć wypełnić luki, usunąć podkreślnik, umieścić przecinka między argumentami, lub zamknąć znak <code>&quot;<\\/code>, <code>'<\\/code>, <code>(<\\/code> lub <code>{<\\/code> odpowiadającym <code>&quot;<\\/code>, <code>'<\\/code>, <code>)<\\/code> lub <code>}<\\/code>.\\n\",\"unparsablequotes\":\"<p>Wygląda na to, że Twój kod zawiera szczególnie sformatowane cudzysłowy lub cudzysłowy typograficzne (<code>{{character}}<\\/code>) przy ciągach znaków, co sprawia, że kod jest niepoprawny. R wymaga cudzysłowów prostych (<code>&quot;<\\/code> albo <code>'<\\/code>).<\\/p> {{code}} <p>Nie martw się, to powszechne źródło błędów, gdy kopiuje się kod z innego programu, który sam formatuje teskt. Możesz spróbować zastąpić swój kod następującym kodem. Mogą być też inne miejsca, które wymagają poprawienia.<\\/p> {{suggestion}}\\n\",\"unparsableunicode\":\"<p>Wygląda na to, że Twój kod zawiera niespodziewany znak specjalny (<code>{{character}}<\\/code>), co sprawia, że kod jest niepoprawny.<\\/p> {{code}} <p>Czasami Twój kod może zawierać znak specjalny, który wygląda jak zwykły znak, zwłaszcza jeśli kopiujesz kod z innego programu. Spróbuj usunąć znak specjalny i wpisać do ponownie ręcznie.<\\/p>\\n\",\"unparsableunicodesuggestion\":\"<p>Wygląda na to, że Twój kod zawiera niespodziewany znak specjalny (<code>{{character}}<\\/code>), co sprawia, że kod jest niepoprawny.<\\/p> {{code}} <p>Czasami Twój kod może zawierać znak specjalny, który wygląda jak zwykły znak, zwłaszcza jeśli kopiujesz kod z innego programu. Możesz spróbować zastąpić swój kod następującym kodem. Mogą być też inne miejsca, które wymagają poprawienia.<\\/p> {{suggestion}}\\n\",\"and\":\"i\",\"or\":\"lub\",\"listcomma\":\", \",\"oxfordcomma\":\"\"}}}}}<\/script>"]},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.10.6.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["tutorial-format"]},{"type":"character","attributes":{},"value":["0.10.6.9000"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmarkdown/templates/tutorial/resources"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["tutorial-format.js"]},{"type":"character","attributes":{},"value":["tutorial-format.css","rstudio-theme.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.10.6.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["jquery"]},{"type":"character","attributes":{},"value":["3.6.0"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/3.6.0"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery-3.6.0.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquerylib"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.1.4"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["navigation"]},{"type":"character","attributes":{},"value":["1.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/navigation-1.1"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["tabsets.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["2.16"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["highlightjs"]},{"type":"character","attributes":{},"value":["9.12.0"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/highlightjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["highlight.js"]},{"type":"character","attributes":{},"value":["default.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["2.16"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["jquery"]},{"type":"character","attributes":{},"value":["3.6.0"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/3.6.0"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery-3.6.0.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquerylib"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.1.4"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["font-awesome"]},{"type":"character","attributes":{},"value":["5.1.0"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/fontawesome"]}]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["css/all.css","css/v4-shims.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["2.16"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["bootbox"]},{"type":"character","attributes":{},"value":["5.5.2"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/bootbox"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["bootbox.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.10.6.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["idb-keyvalue"]},{"type":"character","attributes":{},"value":["3.2.0"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/idb-keyval"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["idb-keyval-iife-compat.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[false]},{"type":"character","attributes":{},"value":["0.10.6.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["tutorial"]},{"type":"character","attributes":{},"value":["0.10.6.9000"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/tutorial"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["tutorial.js"]},{"type":"character","attributes":{},"value":["tutorial.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.10.6.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.4.14"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.10.6.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.10.6.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.4.14"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.10.6.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.10.6.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.4.14"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.10.6.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.10.6.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.4.14"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.10.6.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.10.6.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.4.14"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.10.6.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.10.6.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.4.14"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.10.6.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.10.6.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.4.14"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.10.6.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.10.6.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.4.14"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.10.6.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.10.6.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.4.14"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.10.6.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.10.6.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["ace"]},{"type":"character","attributes":{},"value":["1.4.14"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/ace"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["ace.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.10.6.9000"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["clipboardjs"]},{"type":"character","attributes":{},"value":["2.0.10"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/clipboardjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["clipboard.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.10.6.9000"]}]}]}
</script>
<!--/html_preserve-->
<!--html_preserve-->
<script type="application/shiny-prerendered" data-context="execution_dependencies">
{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["packages"]}},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["packages","version"]},"class":{"type":"character","attributes":{},"value":["data.frame"]},"row.names":{"type":"integer","attributes":{},"value":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61]}},"value":[{"type":"character","attributes":{},"value":["assertthat","backports","base","bslib","cachem","checkmate","cli","compiler","curl","datasets","DBI","digest","dplyr","ellipsis","evaluate","fansi","fastmap","generics","glue","gradethis","graphics","grDevices","htmltools","htmlwidgets","httpuv","jquerylib","jsonlite","knitr","later","learnr","lifecycle","magrittr","markdown","methods","methylCIPHER","mime","pillar","pkgconfig","promises","purrr","R6","Rcpp","rlang","rmarkdown","rprojroot","rstudioapi","sass","shiny","stats","stringi","stringr","tibble","tidyselect","tools","utf8","utils","vctrs","withr","xfun","xtable","yaml"]},{"type":"character","attributes":{},"value":["0.2.1","1.4.1","3.6.1","0.4.0","1.0.6","2.1.0","3.3.0","3.6.1","4.3.2","3.6.1","1.1.2","0.6.29","1.0.9","0.3.2","0.16","1.0.3","1.1.0","0.1.2","1.6.2","0.2.10.9000","3.6.1","3.6.1","0.5.3","1.5.4","1.6.5","0.1.4","1.8.0","1.40","1.3.0","0.10.6.9000","1.0.1","2.0.3","1.1","3.6.1","0.2.0","0.12","1.8.1","2.0.3","1.2.0.1","0.3.4","2.5.1","1.0.9","1.0.4","2.16","2.0.3","0.14","0.4.2","1.7.2","3.6.1","1.7.8","1.4.1","3.1.8","1.1.2","3.6.1","1.2.2","3.6.1","0.4.1","2.5.0","0.32","1.8-4","2.3.5"]}]}]}
</script>
<!--/html_preserve-->
</div>
</div>

</article> <!-- topics -->

<div class="topicsContainer">
<div class="topicsPositioner">
<div class="band">
<div class="bandContent topicsListContainer">

<!-- begin doc-metadata -->
<div id="doc-metadata">
<h1 class="title toc-ignore" style="display:none;">methylCIPHER
tutorial</h1>
<h4 class="author"><em>Kyra Thrush</em></h4>
<h4 class="date"><em>2022-08-30</em></h4>
</div>
<!-- end doc-metadata -->

</div> <!-- bandContent.topicsListContainer -->
</div> <!-- band -->
</div> <!-- topicsPositioner -->
</div> <!-- topicsContainer -->


</main> <!-- bandContent page -->
</div> <!-- pageContent band -->



<!-- Build Tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("section-TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<script>
// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});
</script>


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>


</body>

</html>
